pyuvdata/uvdata.py

```diff
@@ -1424,6 +1424,118 @@ def reorder_blts(self, order='time', minor_order=None, conj_convention=None, uvw
             self.check(check_extra=check_extra,
                        run_check_acceptability=run_check_acceptability)
 
+    def sum_vis(self, other, run_check=True, check_extra=True,
+                run_check_acceptability=True, inplace=False, difference=False):
+        """
+        Sums visibilities between two UVData objects.
+
+        Parameters
+        ----------
+        other : UVData object
+            Another UVData object which will be added to self.
+        run_check : bool
+            Option to check for the existence and proper shapes of parameters
+            after combining objects.
+        check_extra : bool
+            Option to check optional parameters as well as required ones.
+        run_check_acceptability : bool
+            Option to check acceptable range of the values of parameters after
+            combining objects.
+        inplace : bool
+            If True, overwrite self as we go, otherwise create a third object
+            as the sum of the two.
+        difference : bool
+            If True, differences the visibilities of the two UVData objects rather than summing them.
+
+        Returns
+        ------
+        UVData Object
+            If inplace parameter is False.
+
+        Raises
+        ------
+        ValueError
+            If other is not a UVData object, or if self and other
+            are not compatible.
+
+        """
+
+        if inplace:
+            this = self
+        else:
+            this = copy.deepcopy(self)
+
+        # Check that both objects are UVData and valid
+        this.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+        if not issubclass(other.__class__, this.__class__):
+            if not issubclass(this.__class__, other.__class__):
+                raise ValueError('Only UVData (or subclass) objects can be '
+                                    'added to a UVData (or subclass) object')
+        other.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+
+        compatibility_params = ['_Nants_data', '_Nants_telescope',
+                                '_Nbls', '_Nblts', '_Nfreqs', '_Npols',
+                                '_Nspws', '_Ntimes', '_ant_1_array',
+                                '_ant_2_array', '_antenna_diameters',
+                                '_antenna_names', '_antenna_numbers',
+                                '_antenna_positions', '_baseline_array',
+                                '_blt_order', '_channel_width',
+                                '_dut1', '_earth_omega',
+                                '_extra_keywords', '_flag_array',
+                                '_freq_array', '_gst0', '_history',
```

Comment:
we probably don't want to require that the histories are the same.

Reply:
I removed the history parameter. To make this list I just grabbed every parameter, minus the data_array and now the history, so let me know if others are unnecessary.

Reply:
I think this is a good set for now. Maybe add a comment above this list to say it's everything except the history and the data_array? Or better yet, make the list this way:
```
compatibility_params = list(temp.__iter__())
compatibility_params.remove('_data_array')
compatibility_params.remove('_history')
```